common/src/IO/MipTextureReader.cpp

```diff
@@ -47,43 +47,53 @@ namespace TrenchBroom {
         Assets::Texture* MipTextureReader::doReadTexture(const char* const begin, const char* const end, const Path& path) const {
             static const size_t MipLevels = 4;
             
-            static Color tempColor, averageColor;
-            static Assets::TextureBuffer::List buffers(MipLevels);
-            static size_t offset[MipLevels];
```

Comment:
Well, there is in the case of buffers. If they are static, then there will much less resizing going on, since the buffers will be reused for every texture.

Reply:
It'll reuse buffers, but that's just a vector of 4 std::shared_ptr's.  It doesn't look like the buffers themselves are reused, currently. `setMipBufferSize` is doing `buffers[level] = Assets::TextureBuffer(numBytes);` for each mip level. That'll always deallocate the previous texture's buffer and allocate a new one, right?

For later, if we want to make this faster I'd run `MipTextureReader::doReadTexture` in parallel, it seems like an ideal + safe thing to parallelize. We could use thread_local to actually reuse the buffers if allocations are a bottleneck.

Reply:
true